eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/steps/VariantsAnnotGenerateInputBatch.java

```diff
@@ -0,0 +1,190 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package embl.ebi.variation.eva.pipeline.steps;
+
+import com.mongodb.DBObject;
+import embl.ebi.variation.eva.pipeline.ConnectionHelper;
+import embl.ebi.variation.eva.pipeline.annotation.generateInput.VariantAnnotationItemProcessor;
+import embl.ebi.variation.eva.pipeline.annotation.generateInput.VariantWrapper;
+import embl.ebi.variation.eva.pipeline.jobs.VariantJobArgsConfig;
+import org.opencb.datastore.core.ObjectMap;
+import org.opencb.opencga.storage.core.variant.VariantStorageManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.core.Step;
+import org.springframework.batch.core.configuration.annotation.EnableBatchProcessing;
+import org.springframework.batch.core.configuration.annotation.StepBuilderFactory;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.batch.item.data.MongoItemReader;
+import org.springframework.batch.item.file.FlatFileItemWriter;
+import org.springframework.batch.item.file.transform.BeanWrapperFieldExtractor;
+import org.springframework.batch.item.file.transform.DelimitedLineAggregator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Diego Poggioli
+ *
+ * Dump a list of variants without annotations from mongo
+ *
+ * Step class that:
+ * - READ: read the variants without annotations from mongo
+ * - PROCESS convert, filter, validate... the {@link VariantWrapper}
+ * - LOAD: write the {@link VariantWrapper} into a flatfile
+ *
+ * TODO:
+ * - Handle the overwrite
+ * - The variant list should be compressed. It is not possible to write into a zipped file with FlatFile item writer
+ *  see jmmut comment at https://github.com/EBIvariation/eva-v2/pull/22
+ *  We can create an extra step to convert the file and remove the nonp-zipped one
+ *  https://www.mkyong.com/java/how-to-compress-a-file-in-gzip-format/
+ *  https://examples.javacodegeeks.com/core-java/io/fileinputstream/compress-a-file-in-gzip-format-in-java/
+ *  http://www.journaldev.com/966/java-gzip-example-compress-and-decompress-file-in-gzip-format-in-java
+ */
+
+@Configuration
+@EnableBatchProcessing
+@Import(VariantJobArgsConfig.class)
+public class VariantsAnnotGenerateInputBatch {
+
```

Comment:
How hard is to make a class inherit from Step directly? I've seen documentation on how to do it using XML, but not Java annotations.